source/CommonJobs/CommonJobs.Application.EvalForm/Commands/RevertEvaluationStateCommand.cs

```diff
@@ -0,0 +1,94 @@
+﻿using CommonJobs.Application.EvalForm.Indexes;
+using CommonJobs.Domain.Evaluations;
+using CommonJobs.Infrastructure.RavenDb;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using static CommonJobs.Application.EvalForm.Helper.RevertEvaluationActionsHelper;
+
+namespace CommonJobs.Application.EvalForm.Commands
+{
+    public class RevertEvaluationStateCommand : Command
+    {
+        private string _evaluatedEmployee;
+        private string _period;
+        private RevertAction _action;
+
+        public RevertEvaluationStateCommand(string period, string evaluatedEmployee, string action)
+        {
+            _period = period;
+            _evaluatedEmployee = evaluatedEmployee;
+            _action = MapToAction(action);
+        }
+
+        public override void Execute()
+        {
+            var evaluationId = EmployeeEvaluation.GenerateEvaluationId(_period, _evaluatedEmployee);
+            var evaluation = RavenSession.Load<EmployeeEvaluation>(evaluationId);
+            var califications = RavenSession.Advanced.LoadStartingWith<EvaluationCalification>(evaluationId + "/").ToList();
+            var isResponsibleEvalFinished = califications.Any(x => x.Owner == CalificationType.Responsible && x.Finished);
+            var isCompanyEvalFinished = califications.Any(x => x.Owner == CalificationType.Company && x.Finished);
+            var isAutoEvalFinished = califications.Any(x => x.Owner == CalificationType.Auto && x.Finished);
+            var isEvaluatorEvalFinished = califications.Any(x => x.Owner == CalificationType.Evaluator && x.Finished);
+
+            var possibleActions = GetPosibleRevertActions(
+                evaluation.Finished,
+                evaluation.ReadyForDevolution,
+                isCompanyEvalFinished,
+                isResponsibleEvalFinished,
+                isAutoEvalFinished,
+                isEvaluatorEvalFinished);
+
+            if (!possibleActions.Contains(_action))
```

Comment:
What do you think about to move _this verification_ and `MapToAction()` logic inside the Helper?

Maybe the helper could be a non-static class, that receive  the parameters in the constructor (isFinished,  isReadyForDevolution, isCompanyEvalFinished, isResponsibleEvalFinished, isAutoEvalFinished, isEvaluatorEvalFinished) with two methods, one to get the complete list and another to validate if an action (string) is available.